prog/vm/github_runner.rb

```diff
@@ -155,6 +168,13 @@ def before_run
     hop_setup_environment
   end
 
+  def concurrency_available?
+    DB.transaction do
+      github_runner.installation.project_dataset.for_update.all
+      github_runner.installation.project.runner_core_limit > github_runner.installation.project.github_installations.sum(&:total_active_runner_cores)
```

Comment:
If runner_core_limit is 300 and total usage is 298 cores, we would allow provisioning of 16 cores here, right?